src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetServiceTest.java

```diff
@@ -53,15 +54,76 @@ public void shouldGenerateApplicantCoverSheet() throws Exception {
         assertThat(document().getUrl(), is(caseDocument.getDocumentUrl()));
     }
 
+    @Test
+    public void shouldGenerateApplicantCoverSheetWithEmptySolicitorAddress() throws Exception {
+        CaseDetails caseDetails = caseDetailsWithEmptySolAddress();
+        coverSheetService.prepareApplicantCoverSheet(caseDetails);
+
+        BulkPrintCoverSheet bulkPrintCoverSheet = (BulkPrintCoverSheet) caseDetails.getData().get("bulkPrintCoverSheet");
+
+        assertThat(bulkPrintCoverSheet.getAddressLine1(), is("50 Victoria Street"));
```

Comment:
Something I usually do also to help. Because here I had to check the fixtures files to make sure each were the right address, but to make it easier, you could name each address like this for example: 33 application solicitor street
So when someone looks at the code, and the title of the function, they will know straight away what the function does. And if there is an error also will be faster to find out :)

Reply:
Changed json to provide more clarity as advised